File: same_day_delivery_client/lib/model/user.model.dart
import 'dart:convert';

class UserModel {
  String? userId;
  String userName;
  String userEmail;
  String userPassword;
  List<dynamic> userPhone;
  String userAddress;
  String userDate;
  String role;

  UserModel({
    this.userId,
    required this.userName,
    required this.userEmail,
    required this.userPassword,
    required this.userPhone,
    required this.userAddress,
    required this.userDate,
    required this.role,
  });

  Map<String, dynamic> toJson() {
    return {
      '_id': userId,
      'fullName': userName,
      'email': userEmail,
      'password': userPassword,
      'phoneNumber': userPhone,
      'address': userAddress,
      'date': userDate,
      'role': role,
    };
  }

  factory UserModel.fromJson(Map<String, dynamic> json) {
    return UserModel(
      userId: json['_id'].toString(),
      userName: json['fullName'],
      userEmail: json['email'],
      userPassword: json['password'],
      userPhone: json['phoneNumber'],
      userAddress: json['address'],
      userDate: json['date'],
      role: json['role'] ?? "user",
    );
  }

  String toJsonString() {
    return '''
    {
      "_id": "$userId",
      "fullName": "$userName",
      "email": "$userEmail",
      "password": "$userPassword",
      "phoneNumber": "${userPhone[0].toString()}",
      "address": "$userAddress",
      "date": "$userDate",
      "role": "$role"
    }
    ''';
  }

  factory UserModel.fromJsonString(String jsonString) {
    final Map<String, dynamic> json =
        jsonDecode(jsonString); // Use jsonDecode to parse the JSON string
    return UserModel(
      userId: json['_id'],
      userName: json['fullName'],
      userEmail: json['email'],
      userPassword: json['password'],
      userPhone: (json["phoneNumber"] != null && json["phoneNumber"].isNotEmpty)
          ? [json["phoneNumber"].toString()]
          : [],
      userAddress: json['address'],
      userDate: json['date'],
      role: json['role'] ?? "user",
    );
  }
}

File: SameDayDelivery-Admin/model/user.model.js
const mongoose = require("mongoose");
const { Schema } = mongoose;

const userSchema = new Schema({
  fullName: {
    type: String,
    required: true,
  },
  password: {
    type: String,
    required: true,
    trim: true,
  },
  email: {
    type: String,
    trim: true,
  },
  phoneNumber: {
    type: [String],
    trim: true,
  },

  role: {
    type: String,
    required: true,
    enum: ["rider", "user", "shopkeeper"],
    default: "user",
  },

  address: {
    type: String,
    required: true,
  },
  date: {
    type: Date,
    default: Date.now,
  },
});

const userModel = mongoose.model("users", userSchema);
module.exports = userModel;
